pkg/sign/timestamping.go

```diff
@@ -50,25 +51,27 @@ func (ta *TimestampAuthority) GetTimestamp(b *protobundle.Bundle) error {
 	}
 
 	messageSignature := b.GetMessageSignature()
+	dsseEnvelope := b.GetDsseEnvelope()
 
-	if messageSignature != nil {
+	switch {
+	case messageSignature != nil:
 		signatureHash := sha256.Sum256(messageSignature.Signature)
 		req.HashAlgorithm = crypto.SHA256
 		req.HashedMessage = signatureHash[:]
+	case dsseEnvelope != nil:
+		signatureHash := sha256.Sum256(dsseEnvelope.Signatures[0].Sig)
```

Comment:
How about countersign all signatures? The verification would be a bit tricker as you would need to perform N^2 verifications attempts to match timestamp signature against DSSE signature. But N is small so in practice it won't matter.

Reply:
What would be the motivation over countersigning each over a concatenation of all? In either case, you're committing to the signatures. The latter does mean we need canonicalization, but would reduce verification cost.

BTW, we can punt this from this PR to avoid this getting larger, lets just follow up on an issue and explicitly reject multi-sig DSSEs.

Reply:
I based this off of how sigstore-go handles verification today: https://github.com/sigstore/sigstore-go/blob/75b405ded9aee87afa3c29facd08effbd35e43b2/pkg/bundle/signature_content.go#L100

We can change this, but if we decide to do so, we should probably change it in both places.

Reply:
Can you file an issue to track this? This is worth standardizing across clients too.